### PR TITLE
json backup status, templated backup status output

### PR DESCRIPTION
**Pull request by xcezzz**
--json now works during a backup job and will stream out the status on same interval as the regular output

outputs json information with newlines
```json
{
	"type": "archive",
	"duration": 332406583,
	"directories": 28,
	"files": 324,
	"completed": 130878418,
	"total": 11437998853,
	"bps": 0,
	"completed_bytes": 130878418,
	"items_done": 106,
	"items_total": 352,
	"total_bytes": 11437998853,
	"eta": 0,
	"errors": 0,
	"seconds": 0
}
```

### CODE REVIEW

**Comment by fd0** [comment]
Hey, thanks for your contribution. Unfortunately you didn't open an issue first, so we did not have the chance to talk this through. I did not find the time yet to review the code, please stay tuned!

**Comment by xcezzz** [comment]
Apologies, this was going off of the issue #1063 with the feature request for json output during the backup command. I am open to suggestions. I had used this to integrate restic into a frontend GUI app.

**Comment by fd0** [comment]
Hey, just a quick question: We're switching to the idiomatic Go repository layout (see #1116). May I convert your changes and force-push your branch?

**Comment by fd0** [comment]
Hey, may I move your changes over to the new git repo layout?

**Comment by fd0** [comment]
Or just do `git rebase -i master`, squash the commits and force-push the branch. It should be merged automatically.

File: src/restic/progress.go [hunk 1]
```diff
@@ -43,7 +43,6 @@ type Stat struct {
 	Errors uint64
 }
 
-// ProgressFunc is used to report progress back to the user.
```
Comment (fd0) [inline src/restic/progress.go]: Why did you remove this comment?

Comment (xcezzz) [inline src/restic/progress.go]: it may have been an accident. 
I am not well versed with git. if you wish to move it over to new git repo layout you may go ahead.
